tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/mapper/CollationMapper.java

```diff
@@ -68,88 +71,76 @@
     public static IcuData process(
         IcuData icuData, CldrData cldrData, Optional<CldrData> icuSpecialData, String cldrVersion) {
 
-        CollationVisitor visitor = new CollationVisitor(icuData, cldrVersion);
-        icuSpecialData.ifPresent(s -> s.accept(DTD, visitor));
-        cldrData.accept(DTD, visitor);
-        return visitor.icuData;
+        CollationMapper mapper = new CollationMapper(icuData, cldrVersion);
+        icuSpecialData.ifPresent(specialData -> CLDR_PROCESSOR.process(specialData, mapper, DTD));
+        CLDR_PROCESSOR.process(cldrData, mapper, DTD);
+        return icuData;
     }
 
-    final static class CollationVisitor implements PrefixVisitor {
```

Comment:
Note that in the new code there's no need to directly implement the path visitors anymore. You just bind paths and actions together and let the processor handle visitation. This lets us remove this class and directly implement the mutable collection state in the outer mapper.